Add metrics collection and metrics-based dashboards info to RabbitMQ

yasar-sumologic opened the pull request:
…Otel

## Purpose of this pull request

Changes to rabbitmq-otel app doc after changes in it to add metric collection and metric-based dashboards


## Select the type of change
<!-- What types of changes does your code introduce? Select the checkbox after clicking "Create pull request" button. -->

- [ ] Minor Changes - Typos, formatting, slight revisions
- [X] Update Content - Revisions, updating sections
- [ ] New Content - New features, sections, pages, tutorials
- [ ] Site and Tools - .clabot, version updates, maintenance, dependencies, new packages for the site (Docusaurus, Gatsby, React, etc.)

## Ticket (if applicable)

<!-- enter your Jira, Asana, or GitHub ticket number -->

kimsauce commented on docs/integrations/containers-orchestration/opentelemetry/rabbitmq-opentelemetry.md:
```diff
@@ -11,25 +11,35 @@ import TabItem from '@theme/TabItem';
 <img src={useBaseUrl('img/integrations/containers-orchestration/rabbitmq.png')} alt="icon" width="45"/> <img src={useBaseUrl('img/send-data/otel-color.svg')} alt="Thumbnail icon" width="45"/>
 
-The [RabbitMQ](https://www.rabbitmq.com/getstarted.html) app is a unified log app. Preconfigured dashboards provide insight into error logs. RabbitMQ logs are sent to Sumo Logic through OpenTelemetry [filelog receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/filelogreceiver).
+The [RabbitMQ](https://www.rabbitmq.com/getstarted.html) app is a unified log app. Preconfigured dashboards provide insight into error logs and metrics. RabbitMQ logs and metrics are sent to Sumo Logic through OpenTelemetry [filelog receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/filelogreceiver).
```
```suggestion
The Sumo Logic app for [RabbitMQ](https://www.rabbitmq.com/getstarted.html) is a unified log solution that includes preconfigured dashboards offering insights into error logs and metrics. RabbitMQ logs and metrics are transmitted to Sumo Logic via the OpenTelemetry [filelog receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/filelogreceiver).
```

yasar-sumologic replied:
fixed it

kimsauce commented on docs/integrations/containers-orchestration/opentelemetry/rabbitmq-opentelemetry.md:
```diff
@@ -11,25 +11,35 @@ import TabItem from '@theme/TabItem';
+* **`sumo.datasource`**. Fixed value of **rabbitmq**.
+* **`messaging.system`**. Fixed value of **rabbitmq**.
+* **`deployment.environment`**. This is a user-configured field set at the time of collector installation. It identifies the environment where the rabbitmq env resides, such as `dev`, `prod`, or `qa`.
+* **`messaging.cluster.name`**. User configured. Enter a name to uniquely identify your Rabbitmq cluster. This cluster name will be shown in the Sumo Logic dashboards.
```
```suggestion
* **`messaging.cluster.name`**. User configured. Enter a name to uniquely identify your RabbitMQ cluster. This cluster name will be shown in the Sumo Logic dashboards.
```

yasar-sumologic replied:
fixed it.

kimsauce commented on docs/integrations/containers-orchestration/opentelemetry/rabbitmq-opentelemetry.md:
```diff
@@ -11,25 +11,35 @@ import TabItem from '@theme/TabItem';
 
+### For metrics collection
+
+This RabbitMQ receiver fetches stats from a RabbitMQ node using the [RabbitMQ Management Plugin](https://www.rabbitmq.com/docs/management) and converts them to OpenTelemetry metrics
```
```suggestion
This RabbitMQ receiver fetches stats from a RabbitMQ node using the [RabbitMQ Management Plugin](https://www.rabbitmq.com/docs/management) and converts them to OpenTelemetry metrics.
```

yasar-sumologic replied:
fixed it.

kimsauce commented on docs/integrations/containers-orchestration/opentelemetry/rabbitmq-opentelemetry.md:
```diff
@@ -11,25 +11,35 @@ import TabItem from '@theme/TabItem';
-1. **Configure logging in RabbitMQ**. RabbitMQ supports logging via the following methods: local text log files, syslog and stdout. RabbitMQ logs have six levels of verbosity: debug, info, warning, error, critical, none. For details please visit this [page](https://www.rabbitmq.com/logging.html#log-levels). For the dashboards to work properly, log level needs to be set to **debug**. Default log l ...
-2. **Configure RabbitMQ to write log lines to a local file**. By default, RabbitMQ logs are stored in `/var/log/rabbitmq/rabbit@<hostname>.log`. The default directory for log files is listed in the `RabbitMQ.conf` file. To configure the log output destination to a log file, use one of the following settings, either in the [configuration file](https://www.rabbitmq.com/logging.html). Edit or create ...
+
+1. RabbitMQ logs have six levels of verbosity: debug, info, warning, error, critical, none. For details please visit this [page](https://www.rabbitmq.com/logging.html#log-levels). For the dashboards to work properly, log level needs to be set to **debug**. Default log level is **info**. All logging settings are located in [RabbitMQ.conf](https://www.rabbitmq.com/logging.html).
```
```suggestion
1. RabbitMQ logs have six levels of verbosity: debug, info, warning, error, critical, and none. For details, visit the [RabbitMQ documentation](https://www.rabbitmq.com/logging.html#log-levels). For the dashboards to work properly, the log level needs to be set to **debug**. Default log level is **info**. All logging settings are located in [RabbitMQ.conf](https://www.rabbitmq.com/logging.html).
```

yasar-sumologic replied:
fixed it

kimsauce commented on docs/integrations/containers-orchestration/opentelemetry/rabbitmq-opentelemetry.md:
```diff
@@ -74,9 +84,18 @@ import SetupColl from '../../../reuse/apps/opentelemetry/set-up-collector.md';
+- **`endpoint (no default)`**. The hostname and port of the RabbitMQ instance, separated by a colon. (For example: `localhost:15672`.)
+- **RabbitMQ logs Path**. Enter the path to the log file for your RabbitMQ instance.
+
+The log file path configured to capture rabbitmq logs must be given here. The files are typically located in `/var/log/rabbitmq/rabbit@<hostname>.log`. If you are using a customized path, check the [`rabbitmq.conf`](https://www.rabbitmq.com/logging.html) file for this information.
```
```suggestion
The log file path configured to capture RabbitMQ logs must be given here. The files are typically located in `/var/log/rabbitmq/rabbit@<hostname>.log`. If you are using a customized path, check the [`rabbitmq.conf`](https://www.rabbitmq.com/logging.html) file for this information.
```

yasar-sumologic replied:
fixed it.

sumoanema commented on docs/integrations/containers-orchestration/opentelemetry/rabbitmq-opentelemetry.md:
```diff
@@ -11,25 +11,35 @@ import TabItem from '@theme/TabItem';
 <img src={useBaseUrl('img/integrations/containers-orchestration/rabbitmq.png')} alt="icon" width="45"/> <img src={useBaseUrl('img/send-data/otel-color.svg')} alt="Thumbnail icon" width="45"/>
 
-The [RabbitMQ](https://www.rabbitmq.com/getstarted.html) app is a unified log app. Preconfigured dashboards provide insight into error logs. RabbitMQ logs are sent to Sumo Logic through OpenTelemetry [filelog receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/filelogreceiver).
+The Sumo Logic app for [RabbitMQ](https://www.rabbitmq.com/getstarted.html) is a unified log solution that includes preconfigured dashboards offering insights into error logs and metrics. RabbitMQ logs and metrics are transmitted to Sumo Logic via the OpenTelemetry [filelog receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/filelogreceiver).
```
this is incorrect. Logs are through filelog receiver and metric will be through rabbit mq receiver

yasar-sumologic replied:
fixed it.

sumoanema commented on docs/integrations/containers-orchestration/opentelemetry/rabbitmq-opentelemetry.md:
```diff
@@ -11,25 +11,35 @@ import TabItem from '@theme/TabItem';
 
+### For metrics collection
+
+This RabbitMQ receiver fetches stats from a RabbitMQ node using the [RabbitMQ Management Plugin](https://www.rabbitmq.com/docs/management) and converts them to OpenTelemetry metrics.
```
Can you copy the whole prerequisite section from rabbitmq receiver doc : https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/rabbitmqreceiver#prerequisites

yasar-sumologic replied:
updated it

sumoanema commented on docs/integrations/containers-orchestration/opentelemetry/rabbitmq-opentelemetry.md:
```diff
@@ -74,9 +84,18 @@ import SetupColl from '../../../reuse/apps/opentelemetry/set-up-collector.md';
-The files are typically located in `/var/log/rabbitmq/rabbit@<hostname>.log`. You can add any custom fields which you want to tag along with the data ingested in sumo. Click on the **Download YAML File** button to get the yaml file.
+Below are the inputs required:
+
+- **`endpoint (no default)`**. The hostname and port of the RabbitMQ instance, separated by a colon. (For example: `localhost:15672`.)
```
I see in the rabbitmq doc that username and password are required parameters. But here they are not listed. Are we not using them in them in the input configuration ?

yasar-sumologic replied:
Added username and password as required parameters.

yasar-sumologic commented:
Hi @jpipkin1 / @JV0812 / @mafsumo / @kimsauce 
I've addressed your feedback on PR and believe it's ready for re-review; could you please take another look?

JV0812 commented on docs/integrations/containers-orchestration/opentelemetry/rabbitmq-opentelemetry.md:
```diff
@@ -11,25 +11,38 @@ import TabItem from '@theme/TabItem';
+### For metrics collection
+
+The RabbitMQ receiver supports RabbitMQ versions `3.8` and `3.9`.
+The RabbitMQ Management Plugin must be enabled by following the [official instructions](https://www.rabbitmq.com/management.html#getting-started).
```
```suggestion
The RabbitMQ management plugin must be enabled by following the [official instructions](https://www.rabbitmq.com/management.html#getting-started).
```

JV0812 commented on docs/integrations/containers-orchestration/opentelemetry/rabbitmq-opentelemetry.md:
```diff
@@ -11,25 +11,38 @@ import TabItem from '@theme/TabItem';
 
+### For metrics collection
+
+The RabbitMQ receiver supports RabbitMQ versions `3.8` and `3.9`.
```
```suggestion
:::note
The RabbitMQ receiver supports RabbitMQ versions `3.8` and `3.9`.
:::
```

JV0812 commented on docs/integrations/containers-orchestration/opentelemetry/rabbitmq-opentelemetry.md:
```diff
@@ -11,25 +11,38 @@ import TabItem from '@theme/TabItem';
+The RabbitMQ receiver supports RabbitMQ versions `3.8` and `3.9`.
+The RabbitMQ Management Plugin must be enabled by following the [official instructions](https://www.rabbitmq.com/management.html#getting-started).
+
+Also, a user with at least [monitoring](https://www.rabbitmq.com/management.html#permissions) level permissions must be used for monitoring.
```
```suggestion
Also, make sure you at least have the [monitoring](https://www.rabbitmq.com/management.html#permissions) level permissions to monitor the metrics.
```

JV0812 commented on docs/integrations/containers-orchestration/opentelemetry/rabbitmq-opentelemetry.md:
```diff
@@ -11,25 +11,38 @@ import TabItem from '@theme/TabItem';
-1. **Configure logging in RabbitMQ**. RabbitMQ supports logging via the following methods: local text log files, syslog and stdout. RabbitMQ logs have six levels of verbosity: debug, info, warning, error, critical, none. For details please visit this [page](https://www.rabbitmq.com/logging.html#log-levels). For the dashboards to work properly, log level needs to be set to **debug**. Default log l ...
-2. **Configure RabbitMQ to write log lines to a local file**. By default, RabbitMQ logs are stored in `/var/log/rabbitmq/rabbit@<hostname>.log`. The default directory for log files is listed in the `RabbitMQ.conf` file. To configure the log output destination to a log file, use one of the following settings, either in the [configuration file](https://www.rabbitmq.com/logging.html). Edit or create ...
+
+1. RabbitMQ logs have six levels of verbosity: debug, info, warning, error, critical, and none. For details, visit the [RabbitMQ documentation](https://www.rabbitmq.com/logging.html#log-levels). For the dashboards to work properly, the log level needs to be set to **debug**. Default log level is **info**. All logging settings are located in [RabbitMQ.conf](https://www.rabbitmq.com/logging.html).
```
```suggestion
1. **RabbitMQ logs have six levels of verbosity**. Debug, info, warning, error, critical, and none. For more information, refer to the [RabbitMQ documentation](https://www.rabbitmq.com/logging.html#log-levels). To make sure the dashboard work properly, the log level needs to be set to **debug**. Default log level is **info**. All logging settings are located in [RabbitMQ.conf](https://www.rabbitmq.com/logging.html).
```

JV0812 commented on docs/integrations/containers-orchestration/opentelemetry/rabbitmq-opentelemetry.md:
```diff
@@ -74,9 +87,22 @@ import SetupColl from '../../../reuse/apps/opentelemetry/set-up-collector.md';
-In this step, we will be configuring the yaml file required for RabbitMQ Collection. Path of the log file configured to capture RabbitMQ logs is needed to be given here.
+OpenTelemetry works with a [configuration](https://opentelemetry.io/docs/collector/configuration/) yaml file with all the details concerning the data that needs to be collected. For example, it specifies the location of a log file that is read and sent to the Sumo Logic platform.
+
+In this step, you will configure the yaml file required for RabbitMQ Collection.
```
```suggestion
In this step, you will configure the yaml file required for RabbitMQ collection.
```

JV0812 commented on docs/integrations/containers-orchestration/opentelemetry/rabbitmq-opentelemetry.md:
```diff
@@ -173,10 +201,26 @@ This sample Query is from the **RabbitMQ - Logs dashboard** > **Events** by Seve
 
+### Metrics
+
+This sample query is from the **RabbitMQ - Metrics dashboard** > **Average Number of Consumers**  panel.
```
```suggestion
This sample query is from the **Average Number of Consumers** panel in **RabbitMQ - Metrics** dashboard.
```

JV0812 commented on docs/integrations/containers-orchestration/opentelemetry/rabbitmq-opentelemetry.md:
```diff
@@ -173,10 +201,26 @@ This sample Query is from the **RabbitMQ - Logs dashboard** > **Events** by Seve
+### Metrics
 
-The **RabbitMQ - Overview** dashboard gives you an at-a-glance view of Error messages, error by severity, top and last 10 errors, Broker and Event Start/Add log messages.
+The **RabbitMQ - Metrics** dashboard gives you an at-a-glance view of your RabbitMQ deployment across brokers, queue, exchange, consumer, messages.
```
```suggestion
The **RabbitMQ - Metrics** dashboard gives you an at-a-glance view of your RabbitMQ deployment across brokers, queue, exchange, consumer, and messages.
```

JV0812 commented on docs/integrations/containers-orchestration/opentelemetry/rabbitmq-opentelemetry.md:
```diff
@@ -173,10 +201,26 @@ This sample Query is from the **RabbitMQ - Logs dashboard** > **Events** by Seve
-### Overview
+### Logs
+
+The **RabbitMQ - Logs** dashboard gives you an at-a-glance view of Error messages, error by severity, top and last 10 errors, Broker and Event Start/Add log messages.
```
```suggestion
The **RabbitMQ - Logs** dashboard gives you an at-a-glance view of error messages, error by severity, top and last 10 errors, broker, and Event Start/Add log messages.
```

JV0812 reviewed (changes requested):
Minor fixes

JV0812 reviewed (approved):
LGTM

jpipkin1 commented:
@yasar-sumologic, can we publish this PR now, or do we need to wait for a release?
